pkg/policy/mapstate.go

```diff
@@ -725,6 +725,12 @@ func (ms *mapState) insertWithChanges(newKey Key, newEntry mapStateEntry, featur
 			}
 		}
 	} else {
+		// authPreferredInsert takes care for precedence and auth
+		if features.contains(authRules) {
```

Comment:
Does it really make sense to skip the precedence logic separately if auth is enabled? I don't love the changing.

I'd rather see the BroaderOrEqualKeys iteration pass to some kind of `mergeParent()` function that can either merge auth from above or prevent insertion of the child element in the event the parent has higher precedence.

Reply:
Without this the main business logic here became much more complicated and much harder to read. That's why I decided to keep the added complexity in `authPreferredInsert()`, even when it now needs to also implement the proxy port precedence logic. There was no clean way to separate proxy port precedence logic from auth propagation logic any more, and given that the non-auth code path is actually used in production IMO this is the right choice.

Reply:
Auth propagation also uses a bit more expensive iteration that iterates in LPM order that is not needed for the non-auth case. So really `authPreferredInsert()` implements the generic logic for adding an allow entry now, and the non-auth variant in `insertWithChanges` is a faster implementation for the most common case.